demo/nodejs/example.js

```diff
@@ -20,45 +21,59 @@ const issueData = {
     ],
     invoice: "rgb20:outpoint1mzu8vz3jly3rzzkdpph583yahv9wktljtfcln6pe2le6n7ehqulstu967t?amount=5&asset=rgb:id1yqqqxya60n725eszngdx8yvwh3pxyk0sp9fszmzxze3nzhgm76ur4dqf2f7gy",
     prototype_psbt: "cHNidP8BAFICAAAAAZ38ZijCbFiZ/hvT3DOGZb/VXXraEPYiCXPfLTht7BJ2AQAAAAD/////AfA9zR0AAAAAFgAUezoAv9wU0neVwrdJAdCdpu8TNXkAAAAATwEENYfPAto/0AiAAAAAlwSLGtBEWx7IJ1UXcnyHtOTrwYogP/oPlMAVZr046QADUbdDiH7h1A3DKmBDck8tZFmztaTXPa7I+64EcvO8Q+IM2QxqT64AAIAAAACATwEENYfPAto/0AiAAAABuQRSQnE5zXjCz/JES+NTzVhgXj5RMoXlKLQH+uP2FzUD0wpel8itvFV9rCrZp+OcFyLrrGnmaLbyZnzB1nHIPKsM2QxqT64AAIABAACAAAEBKwBlzR0AAAAAIgAgLFSGEmxJeAeagU4TcV1l82RZ5NbMre0mbQUIZFuvpjIBBUdSIQKdoSzbWyNWkrkVNq/v5ckcOrlHPY5DtTODarRWKZyIcSEDNys0I07Xz5wf6l0F1EFVeSe+lUKxYusC4ass6AIkwAtSriIGAp2hLNtbI1aSuRU2r+/lyRw6uUc9jkO1M4NqtFYpnIhxENkMak+uAACAAAAAgAAAAAAiBgM3KzQjTtfPnB/qXQXUQVV5J76VQrFi6wLhqyzoAiTACxDZDGpPrgAAgAEAAIAAAAAAACICA57/H1R6HV+S36K6evaslxpL0DukpzSwMVaiVritOh75EO3kXMUAAACAAAAAgAEAAIAA",
-    consignment_file: "/tmp/rgb-node/output/consignment",
+    consignment_file: consignmentPath,
     transaction_file: "/tmp/rgb-node/output/transaction"
 }
 
 var runtime = null
 
 async function main() {
-    await ex.start_rgb(
-        config.network, config.stash_endpoint, JSON.stringify(config.contract_endpoints), config.threaded, config.datadir)
+    await rgbNode.startRgb(
+        config.network, config.stash_endpoint, config.contract_endpoints, config.threaded, config.datadir)
     .then(r => {
+        console.log("RGB node runtime has started")
         runtime = r
-        return ex.issue(runtime, JSON.stringify(issueData))
+        return rgbNode.issue(runtime, issueData)
     })
     /*
     .then(() => {
-        return ex.transfer(runtime, JSON.stringify(transferData.inputs), JSON.stringify(transferData.allocate),
+        return rgbNode.transfer(runtime, transferData.inputs, transferData.allocate,
            transferData.invoice, transferData.prototype_psbt, transferData.consignment_file,
            transferData.transaction_file)
     })
     .then(() => {
-        return ex.asset_allocations(runtime, 'rgb1w82xuaxz6lp9symrp3f4r47rylkkxsh506qzkt2n2kjfhrhrt03qrrcm0g')
+        return rgbNode.assetAllocations(runtime, 'rgb1w82xuaxz6lp9symrp3f4r47rylkkxsh506qzkt2n2kjfhrhrt03qrrcm0g')
     })
+    .then(allocations => {
     */
     .then(() => {
-        return ex.outpoint_assets(runtime, '5aa2d0a8098371ee12b4b59f43ffe6a2de637341258af65936a5baa01da49e9b:0')
+        //console.log("Allocations: " + allocations)
+        console.log("Querying assets")
+        return rgbNode.outpointAssets(runtime, inputOutpoint)
     })
     .then(assets => {
-        console.log('assets: ' + assets)
+        console.log("Asset list for '" + inputOutpoint + "': " + assets)
+        console.log(assets)
+        return rgbNode.validate(runtime, consignmentPath)
+    })
+    .then(() => {
+        console.log("Validation succeded")
```

Comment:
Here the validation may fail; validation method returns result which can also be a set of messages about what is failed. Is it possible to display the result here?

Reply:
the `validate` method returns a `Result<(), RequestError>`, which is translated, by swig, into a resolved `Promise` (which makes code continue to the next `then` block) or into a rejected one (which makes code go to the first `catch` block). so since the succesful response in rust is an empty one, `()`, the successive `then` block will not receive any useful response object, and if you try receiving a response and log the object you will see a `COpaqueStruct` pointing to nothing. in case of an unsuccessful response rust will return a `RequestError`, which currently is catched by the only `catch` block there is in this demo, which will print out validation error (or any other error occurs during the chain).

we can improve this and make a separated chain of `Promise`s for the validation procedure. but I'd prefer to do that in the next PR since I will need to touch the demo soon again

Reply:
Oh ok, it throws exception on rust `Result::Err` ...